### PR TITLE
Rework the README to put the tutorial front and center.

### PR DESCRIPTION
**Pull request by MarcusSorealheis**
here is an attempt to rework the README

### CODE REVIEW

**Comment by MarcusSorealheis** [comment]
image

File: README.md [hunk 1]
```diff
@@ -1,8 +1,107 @@
...
+  <vector_dimension>
+  <query_file>
+  <commit_at_number_of_documents>
+  <topK>
+  <no. of HNSW indexing threads>
+  <no. of cuvs indexing threads>
+  <merge_strategy options: NO_MERGE | TRIVIAL_MERGE | NON_TRIVIAL_MERGE>
+  <queryThreads>
+  <hnswMaxConn>
+  <hnswBeamWidth>
```
Comment (aaronmondal) [inline README.md]: This is obsolete. It's all part of the bazel invocation.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,107 @@
...
+Java(TM) SE Runtime Environment (build 21.0.3+7-LTS-152)
+Java HotSpot(TM) 64-Bit Server VM (build 21.0.3+7-LTS-152, mixed mode, sharing)
+```
+8. wget https://dlcdn.apache.org/maven/maven-3/3.9.8/binaries/apache-maven-3.9.8-bin.tar.gz
+9. tar -xvf apache-maven-3.9.8-bin.tar.gz
+10. mv apache-maven-3.9.8 /opt/
+11. M2_HOME=‘/opt/apache-maven-3.9.8'
+12. PATH="$M2_HOME/bin:$PATH"
+13. export PATH
+14. mvn -version
```
Comment (aaronmondal) [inline README.md]: This section is obsolete. The way to set up an instance now is the setup in the 'Setup' section below. If you want a copy-paste variant, it probably looks something like this:

```bash
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install
# TODO there will be some command printed after the installation to refresh the shell.
git clone git clone https://github.com/TraceMachina/lucene-on-gpu
cd lucene on gpu
nix develop
bazel run lucene
```

**Review by aaronmondal — CHANGES_REQUESTED** [review]
The image and the brev approach are super nice!

- Double-check that the image is actually rendering correctly. In your fork it's currently not displayed (viewed on a mac/chrome)
- The buildsystem changes obsolete most of the previous setup and need to be adjusted.
- It looks like there are some more leftover pointers to the old repo in this readme. We should probably remove those.

File: README.md (hunk 1 above)
Comment (MarcusSorealheis) [reply README.md]: can you link me to the new changes? That would make sense. I will test them.

File: README.md (hunk 2 above)
Comment (MarcusSorealheis) [reply README.md]: oh wow

Comment (MarcusSorealheis) [reply README.md]: are the steps for indexing data still relevant.

Comment (aaronmondal) [reply README.md]: Nah it's all automated :P

If you take a look at the commands, it's basically all just environment setup for nix. Once you have that the only thing that needs to be run is the `bazel run lucene` command which prepares all the dependencies for this shellscript:

https://github.com/TraceMachina/lucene-on-gpu/blob/main/lucene/demo.sh

Building the executable in this script and the fetching the dataset etc all happen as part of the invocation.

The nix environment prepares CUDA, then some custom Bazel overlays sparsely build the part of CuVS that is required. The only thing that you might need to change, depending on your GPU model is this line here which specifies the GPU target architecture:

https://github.com/TraceMachina/lucene-on-gpu/blob/ed82e370ed2c8ce43e5ada2fd7cd3d98654de35b/thirdparty/global_defs.bzl#L6

IIRC the code likely works on GPUs newer than this architecture, but it won't work on older GPUs. Also, the lower you set this the less "specific" or "modern" optimizations you'll get. So for best performance you'd want to set this to exactly the architecture of the GPU that you intend to run it on. Here is a list:

https://developer.nvidia.com/cuda-gpus

File: README.md (hunk 1 above)
Comment (aaronmondal) [reply README.md]: The instructions are in the README.

Comment (aaronmondal) [reply README.md]: https://github.com/TraceMachina/lucene-on-gpu?tab=readme-ov-file#%EF%B8%8F-setup

**Comment by aaronmondal** [comment]
LGTM!

Note that we'll need to update the docs on the specific setup of the brev nodes a bit. It'll only work on Ubuntu 24.04 images. I believe even a "raw" 24.04 image would work, but I'm still testing. In any case, this LGTM and we can fix the brev setup later.

**Pull request merged**